modules/k8s/synapse-ha/main.tf

```diff
@@ -0,0 +1,164 @@
+terraform {
+  required_version = ">= 1.6.6"
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = ">= 0.10.1"
+    }
+  }
+}
+
+resource "juju_model" "synapse" {
+  name = var.juju_model_name
+  config = {
+    juju-http-proxy  = "${var.proxy_scheme}://%{if var.proxy_user != "" && var.proxy_pass != ""}${var.proxy_user}:${var.proxy_pass}@%{endif}${var.proxy_hostname}:${var.proxy_port}"
+    juju-https-proxy = "${var.proxy_scheme}://%{if var.proxy_user != "" && var.proxy_pass != ""}${var.proxy_user}:${var.proxy_pass}@%{endif}${var.proxy_hostname}:${var.proxy_port}"
+    juju-no-proxy    = "127.0.0.1,localhost,::1,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.canonical.com,.launchpad.net,.internal,.jujucharms.com,.ubuntu.com"
+  }
+  constraints = "root-disk-source=volume"
+  cloud {
+    name   = var.cloud_name
+    region = "default"
+  }
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+resource "juju_application" "synapse" {
+  name  = var.synapse_application_name
+  model = var.juju_model_name
+  trust = true
+
+  charm {
+    name     = "synapse"
+    channel  = var.synapse_charm_channel
+    revision = var.synapse_charm_revision
+    series   = var.synapse_charm_series
+  }
+
+  config = {
+    server_name                        = var.synapse_server_name
+    enable_mjolnir                     = var.synapse_enable_mjolnir
+    public_baseurl                     = var.synapse_public_baseurl
+    notif_from                         = var.synapse_notif_from
+    allow_public_rooms_over_federation = var.allow_public_rooms_over_federation
+    backup_passphrase                  = var.synapse_backup_passphrase
+    enable_irc_bridge                  = var.synapse_enable_irc_bridge
+    irc_bridge_admins                  = var.synapse_irc_bridge_admins
+  }
+
+  units = var.synapse_units
+}
+
+resource "juju_application" "nginx_ingress_integrator" {
+  name  = "nginx-ingress-integrator"
+  model = var.juju_model_name
+  trust = true
+
+  charm {
+    name     = "nginx-ingress-integrator"
+    channel  = var.nginx_ingress_integrator_charm_channel
+    revision = var.nginx_ingress_integrator_charm_revision
+  }
+
+  config = {
+    service-hostname = var.synapse_nginx_ingress_integrator_charm_service_hostname
+    tls-secret-name  = var.nginx_ingress_integrator_charm_tls_secret_name
```

Comment:
I remember something about using lego provider instead of setting tls secret.

Reply:
If I recall correctly, the lego provider can create a cert and that one is placed in Vault, so a deployment can reference it. I mean this can stay as is (for now) and in the deployment it is set to an empty string and a lego provider resource together with a relation is created like in here -> https://github.com/canonical/is-prod-synapse-external/blob/main/main.tf